File: java/src/main/java/com/example/backend/adapter/driving/TestCryptoController.java
package com.example.backend.adapter.driving;

import com.example.backend.utility.CryptoUtility;
import com.example.backend.utility.JsonUtility;
import com.fasterxml.jackson.core.type.TypeReference;
import jakarta.servlet.http.HttpServletRequest;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.web.bind.annotation.*;

import java.time.LocalDateTime;
import java.util.Map;

/**
 * 测试加解密 Controller
 */
@RestController
public class TestCryptoController {

    private static final Logger logger = LoggerFactory.getLogger(TestCryptoController.class);

    /**
     * JWT 编码
     */
    @RequestMapping(value = "/api/test/crypto/jwt-encode", method = RequestMethod.POST)
    public Result testCryptoJwtEncode(@RequestBody String requestBody)
    {
        var requestJson = RequestValueHelper.getRequestJson(requestBody);
        String secret = RequestValueHelper.getRequestJsonStringTrimReq(requestJson, "secret");
        LocalDateTime expiresAt = RequestValueHelper.getRequestJsonDateTimeReq(requestJson, "expiresAt");
        var payloadNode = requestJson.path("payload");
        // System.out.println("payloadNode: " + payloadNode.toString());
        Map<String, ?> payload = JsonUtility.deserialize(payloadNode, new TypeReference<Map<String, ?>>() { });
        // System.out.print("payload: " + JsonUtility.serialize(payload));
        String token = CryptoUtility.jwtEncode(payload, secret, expiresAt);
        return Result.okData(Map.of("token", token));
    }

    /**
     * JWT 解码
     */
    @RequestMapping(value = "/api/test/crypto/jwt-decode", method = RequestMethod.POST)
    public Result testCryptoJwtDecode(@RequestBody String requestBody)
    {
        var requestJson = RequestValueHelper.getRequestJson(requestBody);
        String secret = RequestValueHelper.getRequestJsonStringTrimReq(requestJson, "secret");
        String token = RequestValueHelper.getRequestJsonStringTrimReq(requestJson, "token");
        Map<String, ?> payload = CryptoUtility.jwtDecode(token, secret);

        String headerDecoded = "";
        String payloadDecoded = "";
        String[] tokenParts = token.split("\\.");
        if (tokenParts.length > 1) {
            headerDecoded = CryptoUtility.base64Decode(tokenParts[0]);
        }
        if (tokenParts.length > 2) {
            payloadDecoded = CryptoUtility.base64Decode(tokenParts[1]);
        }

        return Result.okData(Map.of(
                "payload", JsonUtility.serialize(payload),
                "headerDecoded", headerDecoded,
                "payloadDecoded", payloadDecoded
        ));
    }

    /**
     * BASE64 编码
     */
    @RequestMapping(value = "/api/test/crypto/base64-encode", method = RequestMethod.GET)
    public Result testCryptoBase64Encode(HttpServletRequest request)
    {
        String text = RequestValueHelper.getRequestParamStringTrimReq(request, "text");
        String base64 = CryptoUtility.base64Encode(text);
        return Result.okData(Map.of("base64", base64));
    }

    /**
     * BASE64 解码
     */
    @RequestMapping(value = "/api/test/crypto/base64-decode", method = RequestMethod.GET)
    public Result testCryptoBase64Decode(HttpServletRequest request)
    {
        String base64 = RequestValueHelper.getRequestParamStringTrimReq(request, "base64");
        String text = CryptoUtility.base64Decode(base64);
        return Result.okData(Map.of("text", text));
    }

    /**
     * MD5 编码
     */
    @RequestMapping(value = "/api/test/crypto/md5-encode", method = RequestMethod.GET)
    public Result testCryptoMd5Encode(HttpServletRequest request)
    {
        String text = RequestValueHelper.getRequestParamStringTrimReq(request, "text");
        String md5 = CryptoUtility.md5Encode(text);
        return Result.okData(Map.of("md5", md5));
    }
}

File: backend-example-dotnet/src/adapter/driving/TestCryptoController.cs
using BackendExample.Utility;

using log4net;

using Microsoft.AspNetCore.Mvc;

namespace BackendExample.Adapter.Driving
{
    /// <summary>
    /// 测试加解密 Controller
    /// </summary>
    public class TestCryptoController : ControllerBase
    {
        private static readonly ILog logger = LogManager.GetLogger(typeof(TestCryptoController));

        /// <summary>
        /// JWT 编码
        /// </summary>
        [HttpPost("/api/test/crypto/jwt-encode")]
        public async Task<Result> TestCryptoJwtEncode()
        {
            var requestJson = await RequestValueHelper.GetRequestJsonAsync(Request);
            string secret = RequestValueHelper.GetRequestJsonStringTrimReq(requestJson, "secret");
            DateTime expiresAt = RequestValueHelper.GetRequestJsonDateTimeReq(requestJson, "expiresAt");
            IDictionary<string, object> payload = JsonUtility.JsonNodeToDictionary(requestJson["payload"]);
            string token = CryptoUtility.JwtEncode(payload, secret, expiresAt);
            return Result.OkData(new { token });
        }

        /// <summary>
        /// JWT 解码
        /// </summary>
        [HttpPost("/api/test/crypto/jwt-decode")]
        public async Task<Result> TestCryptoJwtDecode()
        {
            var requestJson = await RequestValueHelper.GetRequestJsonAsync(Request);
            string secret = RequestValueHelper.GetRequestJsonStringTrimReq(requestJson, "secret");
            string token = RequestValueHelper.GetRequestJsonStringTrimReq(requestJson, "token");
            IDictionary<string, object> payload = CryptoUtility.JwtDecode(token, secret);

            string headerDecoded = string.Empty;
            string payloadDecoded = string.Empty;
            string[] tokenParts = token.Split('.');
            if (tokenParts.Length > 1)
            {
                headerDecoded = CryptoUtility.Base64Decode(tokenParts[0]);
            }
            if (tokenParts.Length > 2)
            {
                payloadDecoded = CryptoUtility.Base64Decode(tokenParts[1]);
            }

            return Result.OkData(new
            {
                payload = JsonUtility.Serialize(payload), // 手动序列化，防止受全局 json 转换影响,
                headerDecoded = headerDecoded,
                payloadDecoded = payloadDecoded
            });
        }

        /// <summary>
        /// BASE64 编码
        /// </summary>
        [HttpGet("/api/test/crypto/base64-encode")]
        public Result TestCryptoBase64Encode()
        {
            string text = RequestValueHelper.GetRequestParamStringTrimReq(Request, "text");
            string base64 = CryptoUtility.Base64Encode(text);
            return Result.OkData(new { base64 });
        }

        /// <summary>
        /// BASE64 解码
        /// </summary>
        [HttpGet("/api/test/crypto/base64-decode")]
        public Result TestCryptoBaseDecode()
        {
            string base64 = RequestValueHelper.GetRequestParamStringTrimReq(Request, "base64");
            string text = CryptoUtility.Base64Decode(base64);
            return Result.OkData(new { text });
        }

        /// <summary>
        /// MD5 编码
        /// </summary>
        [HttpGet("/api/test/crypto/md5-encode")]
        public Result TestCryptoMd5Encode()
        {
            string text = RequestValueHelper.GetRequestParamStringTrimReq(Request, "text");
            string md5 = CryptoUtility.Md5Encode(text);
            return Result.OkData(new { md5 });
        }
    }
}